### PR TITLE
feat(billing): provision metered workflow_runs price on scale subscriptions

### PR DESCRIPTION
**Pull request by Dhruwang**
## What & why

Two coupled gaps in metered workflow-run billing (ENG-1936):

**1. Provisioning.** Attaching a metered price to the Scale **product** in Stripe does not put it on any subscription — a sub's line items come from an explicit list the app builds in `getCatalogItemsForPlan`, which only ever emitted `base` + `responses`. So a `workflow_runs` metered price was never billed on new Scale subs. Worse, because that price is `usage_type: metered`, `getPriceKind` fell through to its metered→`"responses"` fallback and misclassified it, producing two `scale/responses/monthly` matches and throwing in `getSinglePrice` ("found 2"), which breaks the whole billing catalog fetch (checkout, billing page, plan comparison).

Fix: make `workflow_runs` a first-class price kind — classify by `formbricks_price_kind` metadata (unknown metered kinds still excluded so they can't collide); model it as `workflowRunsPrice` on the catalog item via a new `getOptionalSinglePrice` (present on Scale, `null` elsewhere, throws only on ambiguity); provision it as its own metered line item in checkout and plan-change schedules; include its interval in the mixed-interval guard; bump catalog cache `v2`→`v3`.

**2. Usage visibility.** The synced limit `limits.monthly.workflowRuns` was stored and exposed as `monthlyWorkflowRuns` in the entitlements context but never rendered — Scale customers couldn't see workflow usage against their included volume the way responses/workspaces are shown.

Fix: add `getMonthlyOrganizationWorkflowRunCount` (mirrors the responses count — scoped to the org's workspaces over the billing cycle, **excluding dry runs** to match what's metered), thread it through the billing page, and render a workflow-runs `UsageCard`.

**3. Included-volume vs. price consistency (the dangerous one).** The number the card calls "included" and the amount Stripe actually charges were two independent Stripe objects with nothing tying them together — the card read the entitlement-driven `limits.monthly.workflowRuns` while the price's tiers decide the charge. A flat or paid-from-unit-1 metered price would let the card show "42 of 1000 included" while Stripe invoices all 42: silent over-billing behind a reassuring UI. This PR flips billing from under-bill (harmless) to bill-for-real, so this failure mode is new and customer-facing.

Fix: make the **price the single source of truth**. `resolveWorkflowIncludedVolume` derives the included allowance from the price's free first tier; the catalog exposes it as `workflowRunsIncluded` and the card renders *that* number — so what's shown and what's left uncharged are the same object and cannot drift. If a workflow price is not graduated with a free first tier (finite `up_to`), the **workflow item is dropped** (price + included → null): the card hides and new subs skip the workflow line item — workflows just don't bill (today's harmless state) — and a loud `logger.error` fires. This degrades **workflows only**; base/responses checkout and the billing page stay up for every plan/org (base/responses remain fail-loud since they're load-bearing). No false allowance is ever shown.

Note: this wires new subs + plan changes + the usage card. Existing Scale subs still need a one-time backfill of the workflow line item (ops step, not code).

## Linear ticket

Closes ENG-2193 — Scale subscriptions never get a workflow_runs metered item, so workflow-run usage is metered but never billed (catalog provisioning).
Closes ENG-2194 — workflow-run usage has no user-facing surface (billing-page usage card).

Part of the metered-workflows epic ENG-1936.

## How this was tested

- `vitest run modules/ee/billing/lib/stripe-billing-catalog.test.ts` ✅ 8 passed — updated the workflow_runs test to assert provisioning, plus two new edge cases (`getOptionalSinglePrice` → null when absent; throws "found 2" on duplicate).
- `vitest run modules/ee/billing/lib/organization-billing.test.ts` ✅ 70 passed — covers the checkout interval guard + plan-change schedule builder.
- `tsc --noEmit` ✅ on the touched files.
- `pnpm i18n` ✅ — added `unlimited_workflow_runs` to en-US, generated all 14 target locales, scan validation passes (0 missing keys).
- eslint + prettier via lint-staged ✅.

The workflow-runs count query mirrors the existing (untested) `getMonthlyOrganizationResponseCount`; per repo convention these org count helpers are exercised via e2e/manual, not unit tests.

## Breaking changes

None

## QA / Test Plan

**Preconditions / test data**

- Cloud billing configured. On the Scale Stripe product: an **active** metered price with metadata `formbricks_price_kind: workflow_runs`, `formbricks_interval: monthly`, `usage_type: metered`, attached to the `workflow_run_created` meter. The price MUST be **graduated with a free first tier** (`unit_amount: 0` up to the included volume, e.g. `up_to: 1000`) — that free-tier boundary is what the usage card shows as "included".

**How to test**

- [ ] Subscribe a new org to **Scale (monthly)** via checkout → the Stripe subscription has three items: base, responses, workflow_runs.
- [ ] Subscribe a new org to **Scale (yearly)** → base is yearly; responses and workflow_runs are the monthly metered prices.
- [ ] Open **Settings → Billing** for a Scale org → a "Workflow Runs" usage card shows current runs vs the included volume (dry runs not counted), alongside Responses and Workspaces. The included number equals the price's free-tier boundary.
- [ ] **Consistency:** trigger runs inside the included volume (e.g. 5 of 1000) → the upcoming Stripe invoice shows $0 for those runs, matching the card. Cross a paid tier → invoice charges only the overage.
- [ ] Open Billing for a **Pro/Hobby** org (no workflow price) → no Workflow Runs card is shown.
- [ ] Edge: two active workflow_runs prices for the same plan/interval → billing page fails fast with "Expected at most one … found 2".
- [ ] Edge (degrade-only): make the Scale workflow price flat/per-unit or give it a paid first tier → the Workflow Runs card disappears and a `logger.error` ("Invalid workflow_runs price") fires; the billing page, checkout, and plan comparison still work for every plan. No false "included" allowance is shown.

### CODE REVIEW

File: apps/web/modules/ee/billing/lib/stripe-billing-catalog.test.ts [hunk 1]
```diff
@@ -198,11 +198,12 @@ describe("stripe-billing-catalog", () => {
...
-    "excludes a metered price tagged with a non-responses kind (e.g. workflow_runs) so it does not collide with responses",
+    "provisions a workflow_runs metered price as its own line item without colliding with responses",
     async () => {
-      // A second metered price on Scale, tagged workflow_runs (ENG-1936). Without the getPriceKind
-      // guard it would fall through usage_type=metered -> "responses" and produce two matches for
-      // scale/responses/monthly, breaking the billing page.
+      // A second metered price on Scale, tagged workflow_runs (ENG-1936). getPriceKind classifies it
+      // by its metadata kind rather than the usage_type=metered fallback, so it does NOT produce a
+      // second match for scale/responses/monthly (which would throw "found 2"), and it is added to the
+      // subscription as its own metered line item.
```
[the remark this answers is not in the document]
Comment (Dhruwang) [reply apps/web/modules/ee/billing/lib/stripe-billing-catalog.test.ts:249]: Added in 172673f28 — two explicit tests: `getOptionalSinglePrice` returning null when a plan has no workflow_runs price (Scale provisioned with base + responses only), and throwing `Expected at most one Stripe price for scale/workflow_runs/monthly, but found 2` on duplicate prices. Reused a shared workflow-price factory + standard-catalog fixture. Suite: 8 passed.

**Comment by pandeymangg** [comment]
`resolveWorkflowIncludedVolume` throws inside `fetchStripeBillingCatalog`, which resolves `workflow_runs` for hobby/pro/scale with no `try`/`catch` — so one misconfigured workflow price takes down the entire catalog (checkout, billing page, plan comparison) for every plan and org, not just workflows. Since the fallback it's guarding against is simply "workflow runs don't bill" — today's harmless state — failing closed on just the workflow item (`workflowRunsPrice: null` + a loud `logger.error`) would keep the no-false-allowance guarantee without putting the whole checkout funnel behind Stripe config. Worth settling before this lands on `release/5.3`: the catalog is global with a 10-minute TTL and no per-org isolation, so a single typo in a livemode price is globally fatal within one cache cycle.

**Comment by Dhruwang** [comment]
Good catch — fixed in `40f2f41f7`. The validation error is now caught at the workflow-resolution boundary in `fetchStripeBillingCatalog`: a misconfigured workflow price drops the workflow item only (`workflowRunsPrice` + `workflowRunsIncluded` → null) and logs a loud `logger.error`, instead of throwing. The card hides and new subs skip the workflow line item (workflows just don't bill — the harmless state), while base/responses checkout and the billing page stay up for every plan/org. Base/responses stay fail-loud (`getSinglePrice` throws) since they're load-bearing on every plan. Test updated to assert the degrade + log path (catalog resolves, workflow item null, no line item, rest intact); runbook + PR body updated too.

**Review by pandeymangg — APPROVED** [review]
